library/sha256.c

```diff
@@ -367,17 +382,33 @@ int mbedtls_sha256_finish_ret( mbedtls_sha256_context *ctx,
      * Output final state
      */
 
-    for( s_pos = 0, o_pos = 0; s_pos < 7; s_pos++, o_pos += 4 )
     {
-        (void)mbedtls_platform_put_uint32_be( &output[o_pos],
-                                              ctx->state[s_pos] );
-    }
+        uint32_t initval = mbedtls_platform_random_in_range(0xffffffff);
+        uint32_t offset = mbedtls_platform_random_in_range(7);
+
+        for( o_pos = 0; o_pos < 32; o_pos += 4 )
+        {
+            (void)mbedtls_platform_put_uint32_be( &output[o_pos], initval );
```

Comment:
Do we need to use `mbedtls_platform_put_uint32_be` here to set initval?
Could we use just `memset` or even Mbed TLS memset?

Reply:
Done